tests/TestCase.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Tests;
+
+use PHPUnit\Framework\TestCase as BaseTestCase;
+
+abstract class TestCase extends BaseTestCase
+{
+    protected function get(string $uri)
+    {
+        $this->call($uri, 'GET');
+    }
+
+    protected function post(string $uri)
+    {
+        $this->call($uri, 'POST');
+    }
+
+    private function call(string $uri, string $methodType) {
+        $_SERVER['PATH_INFO'] = $uri;
+        $_SERVER['REQUEST_METHOD'] = $methodType;
+        $this->router->get();
```

Comment:
Yes 
when i call this `$this->get('/test');`  i have to run `$this->router->get()` for the route to take place i didn't want to repeat that in each function + it's better to have this 
`$this->get('/');`
than this 
```
$this->get('/');
$this->router->get();
```
right ?
+ as `router` is defined in `RouterTest` this will work
but maybe i should move this `$this->router = new Router([], new Request);` to the testCase so this will be available in any other test class ? you mentioned that overwriting global variables is a bad practice so this whole approach is wrong anyway why i bother myself thinking about this :D

Reply:
Look above, I think it should be another way around! 🤣 

There's nothing wrong in calling `$this->router->get()` every time you need it, it makes your test very "talkative" about what's going on there, and it's good. 

But if you wish, you can shorten this inside `RouterTest`, having private methods `get()` and `post()` there. No other unit test outside of `RouterTest` should have `router` property, why would you want it to be available across them?